includes/settings/class-google-settings.php

```diff
@@ -78,13 +78,14 @@ public static function register_settings() {
 
 	/**
 	 * Ensure Google tracking ID is of the correct format.
+	 * Allow Measurement ID and legacy Univeral Analytics Tracking ID formats.
 	 *
 	 * @param array $tracking_id Google tracking ID.
 	 *
 	 * @return string
 	 */
 	public static function validate_tracking_id( $tracking_id ) {
-		if ( preg_match( '/^UA-\d{4,9}-\d{1,4}$/', $tracking_id ) ) {
+		if ( preg_match( '/^G-[A-Z0-9]+$/', $tracking_id ) || preg_match( '/^UA-\d{4,9}-\d{1,4}$/', $tracking_id ) ) {
```

Comment:
```suggestion
		if ( preg_match( '/^(G-[A-Z0-9]{10})||(UA-\d{4,9}-\d{1,4})$/', $tracking_id ) ) {
```

Reply:
You can do this with a single `preg_match()` call. Also added the length requirement for GA4 IDs.

Reply:
Nice!